src/html5-qrcode.ts

```diff
@@ -502,6 +511,63 @@ export class Html5Qrcode {
         });
     }
 
+    /**
+     * Checks if current video track has flash light capability.
+     * 
+     * This feature requires that camera is started.
+     *
+     * @returns Promise with detected flash capability result.
+     */
+    public hasFlash(): Promise<boolean> {
+        if (!("ImageCapture" in window)) {
+            return Promise.resolve(false);
+        }
+
+        const track = this.videoElement?.srcObject ? (<MediaStream>this.videoElement.srcObject).getVideoTracks()[0] : null;
+        if (!track) {
+            throw "Scan should be started to detect flash capability";
+        }
+
+        return new ImageCapture(track).getPhotoCapabilities()
+            .then((result) => {
+                return result.fillLightMode.includes("flash");
+            })
+            .catch((err) => {
+                this.logger.warn(err);
+                return false;
+            });
+    }
+
+    /**
+     * Checks if flash is turned on.
+     *
+     * @returns Current flash state.
+     */
+    public isFlashOn(): boolean {
+        return this.flashOn;
+    }
+
+    /**
+     * Change flash state.
+     * 
+     * @param on if true flash will be turned on otherwise turned off.
+     * @returns Promise that flash state has changed.
+     */
+    public setFlash(on: boolean): Promise<void> {
```

Comment:
optional, rename to more readable `shouldSetFlashOn` or something more relevant.